sycl/doc/extensions/proposed/sycl_ext_intel_fpga_datapath.asciidoc

```diff
@@ -0,0 +1,239 @@
+// for syntax highlighting purposes. This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation. All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc. OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 7 specification. All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension does not depend on other SYCL extensions, it is self-contained. Below is a list of interactions with other SYCL extensions. These extensions don't hve to be implemented to support the `sycl_ext_intel_fpga_mem` extension.
+
+* link:sycl_ext_intel_fpga_mem.asciidoc[sycl_ext_intel_fpga_mem.asciidoc] as described in the <<Restrictions on creating fpga_mem objects>> section below
+* link:sycl_ext_oneapi_device_global.asciidoc[sycl_ext_oneapi_device_global.asciidoc] as described in the <<Restrictions on creating fpga_mem objects>> section below
+
+== Status
+Proposed
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized. *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+[NOTE]
+====
+In this document, we use `fpga_datapath` to indicate the proposed `sycl::ext::intel::experimental::fpga_datapath`.
+====
+
+The purpose of this document is to clearly describe and specify `fpga_datapath` and 
+related concepts, types, and mechanisms, and to give examples and context for their usage.
+
+=== Motivation
+`fpga_datapath` is a class template templated on a type T, that represents an object of type T. It is a request to the compiler to implement that object, when present in FPGA device code, in the datapath, as opposed to an off-datapath memory.
+
+=== Example
+[source,c++]
+----
+struct MyClass {
+  bool x;
+};
+
+using namespace sycl::ext::intel::experimental;
+
+sycl::queue q;
+q.single_task([=] {
+  fpga_datapath<int[4]> fm1{1, 3, 5, 7};
+  fpga_datapath<MyClass> fm2;
+  fm2.get().x = fm1[0];
+});
+----
+
+Above we see an example that declares `fpga_datapath` variables `fm1`, `fm2`, and accesses them. 
+`fm1` creates an allocation on the device that is private to the kernel and contains an array of four `int` types that is initialized to `{1, 3, 5, 7}`. `fm2` creates an allocation on the device that is private to the kernel and contains an object of type `MyClass` that is default-initialized.
```

Comment:
fm->fd
also "four 'int' types" sounds off.  That makes it sound like an array of types rather than an array of objects.  
```suggestion
Above we see an example that declares `fpga_datapath` variables `fd1`, `fd2`, and accesses them. 
`fd1` creates an allocation on the device that is private to the kernel and contains an array of four `int`s that is initialized to `{1, 3, 5, 7}`. `fd2` creates an allocation on the device that is private to the kernel and contains an object of type `MyClass` that is default-initialized.
```